lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
+              .then(accept)
+              .catch((err) => {
+                if (numRetries--) {
+                  taskHandler.runtime.log('retrying artifact upload');
+                  setTimeout(retry, taskHandler.runtime.retry.interval);
```

Comment:
nice.  If we change to using a package for retrying here, we should also create an issue to use the same package elsewhere in docker-worker where retrying is done.  Right now the method for retrying and backoff+jitter was inspired (read: copied) from other taskcluster components that do the same.  We should probably standardize how we're doing this across the board.